src/modules/map/components/cesium/CesiumWMTSLayer.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+    <div>
+        <slot />
+    </div>
+</template>
+
+<script>
+import GeoAdminWMTSLayer from '@/api/layers/GeoAdminWMTSLayer.class'
+import { CoordinateSystem, WEBMERCATOR } from '@/utils/coordinateSystems'
+import { ImageryLayer, Rectangle, UrlTemplateImageryProvider } from 'cesium'
+import { TILEGRID_EXTENT_EPSG_4326 } from '@/config'
+import addImageryLayerMixins from './utils/addImageryLayer-mixins'
+
+import { getTimestampFromConfig } from '@/utils/layerUtils'
+
+const MAXIMUM_LEVEL_OF_DETAILS = 18
+
+export default {
+    mixins: [addImageryLayerMixins],
+    props: {
+        wmtsLayerConfig: {
+            type: GeoAdminWMTSLayer,
+            required: true,
+        },
+        previewYear: {
+            type: Number,
+            default: null,
+        },
+        projection: {
+            type: CoordinateSystem,
+            default: WEBMERCATOR,
+        },
+        zIndex: {
+            type: Number,
+            default: -1,
+        },
+    },
+    computed: {
+        layerId() {
+            return this.wmtsLayerConfig.getID()
+        },
+        opacity() {
+            return this.wmtsLayerConfig.opacity || 1.0
+        },
+        timestampForPreviewYear() {
+            return getTimestampFromConfig(this.wmtsLayerConfig, this.previewYear)
+        },
+        url() {
+            return this.wmtsLayerConfig.getURL(
+                this.timestampForPreviewYear,
+                this.projection.epsgNumber
+            )
+        },
+    },
+    methods: {
+        createImagery(url) {
+            return new ImageryLayer(
+                new UrlTemplateImageryProvider({
+                    rectangle: Rectangle.fromDegrees(...TILEGRID_EXTENT_EPSG_4326),
+                    maximumLevel: MAXIMUM_LEVEL_OF_DETAILS,
+                    url: url,
+                }),
+                {
+                    show: this.wmtsLayerConfig.visible,
+                }
+            )
+        },
```

Comment:
`opacity` is not set here, meaning that if I change the opacity of the layer before making it visible, it is ignored

Reply:
opacity will be applied correctly even if the layer is not visible but I added the `alpha` property anyway to be sure that it is always applied correctly